src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -21,11 +22,8 @@ internal abstract partial class AbstractPersistentStorageService(IPersistentStor
 {
     protected readonly IPersistentStorageConfiguration Configuration = configuration;
 
-    /// <summary>
-    /// This lock guards all mutable fields in this type.
-    /// </summary>
     private readonly SemaphoreSlim _lock = new(initialCount: 1);
-    private ReferenceCountedDisposable<IChecksummedPersistentStorage>? _currentPersistentStorage;
+    private readonly ConcurrentDictionary<SolutionKey, IChecksummedPersistentStorage> _solutionKeyToStorage = new();
```

Comment:
switched to a simple map. Storage objects are not expensive.  They tend to just be a a small number of simple handle objects.

Reply:
Maybe it's not to be worried about, but it seems weird to see this not get cleaned up. Wouldn't the single static object work (and still not really be much more complicated than using the dictionary)?

Reply:
Sure.  I can change this.

Reply:
Moved back to storing just the last storage instance.